content/en/docs/tasks/tools/install-kubectl.md

```diff
@@ -295,63 +295,120 @@ kubectl cluster-info dump
 
 ## Enabling shell autocompletion
 
-kubectl includes autocompletion support, which can save a lot of typing!
+kubectl provides autocompletion support for Bash and Zsh, which can save you a lot of typing!
 
-The completion script itself is generated by kubectl, so you typically just need to invoke it from your profile.
+The procedure to properly set up autocompletion is different for Bash and Zsh. Both cases are explained below. 
 
-Common examples are provided here. For more details, consult `kubectl completion -h`.
+- [Bash](#bash)
+- [Zsh](#zsh)
 
-### On Linux, using bash
-On CentOS Linux, you may need to install the bash-completion package which is not installed by default.
+### Bash
 
-```shell
-yum install bash-completion -y
-```
+The kubectl completion script for Bash can be generated with `kubectl completion bash`. Sourcing this script in your shell enables kubectl autocompletion.
+
+However, the completion script depends on [**bash-completion**](https://github.com/scop/bash-completion) (you can test if you already have bash-completion installed by running the command `type _init_completion`).
+
+The below sections walk you through installing bash-completion and setting up kubectl autocompletion for both Linux and macOS.
+
+#### Bash on Linux
```

Comment:
Could we put the advice for different OSs in selectable tabs? (and, does that approach make sense)?

Reply:
Yeah, that's possible. I quickly tested it, it could make sense.

Reply:
I changed it to selectable tabs now.